### PR TITLE
MINOR HT-28 - Add functionality for managing vehicles 

### PR DESCRIPTION
**Pull request by conor27nash**
Adds Manage Vehicles view, which allows a administrator to view all Vehicles , edit specific Vehicles details, add new Vehicles and delete Vehicles . This also includes some refactoring of the test setup and additional database migrations

### CODE REVIEW

File: HaulThis.Test/Integration/DatabaseSetup.cs [hunk 1]
```diff
@@ -5,16 +5,41 @@ namespace HaulThis.Test.Integration;
 public class DatabaseSetup
 {
     public const string LocalDbString = "Server=(localdb)\\MSSQLLocalDB;Database=master;Integrated Security=true;MultipleActiveResultSets=true;";
+    // public const string LocalDbString = "Server=localhost,1433;Database=master;User Id=sa;Password=YourStrong@Passw0rd;TrustServerCertificate=True;MultipleActiveResultSets=True;";
```
Comment (stewartbebbington) [inline HaulThis.Test/Integration/DatabaseSetup.cs:8]: remove

File: HaulThis.Test/Integration/ManageVehiclesTest.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        _databaseSetup = new DatabaseSetup();
+        _connection = _databaseSetup.DeployDatabase();
+        
+        var loggerFactory = LoggerFactory.Create(loggerBuilder =>
+        {
+            loggerBuilder.AddConsole();
+            loggerBuilder.AddDebug();
+        });
+        ILogger<DatabaseService> logger = loggerFactory.CreateLogger<DatabaseService>();
+        _databaseService = new DatabaseService(_connection, logger);
```
Comment (stewartbebbington) [inline HaulThis.Test/Integration/ManageVehiclesTest.cs:26]: you shouldn't need to do all this, you can just extend this class with `DisposableIntegrationTest` to get the dbService then all you have to pass in here is  `_manageVehiclesService = new ManageVehiclesService(_databaseService);`

Look at ManageEmployeesTest for reference

File: HaulThis/ViewModels/ManageVehiclesViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+[assembly: InternalsVisibleTo("HaulThis.Tests")]
+
+namespace HaulThis.ViewModels;
+
+public class ManageVehiclesViewModel : INotifyPropertyChanged
+{   
+    readonly IManageVehiclesService _ManageVehiclesService;
+
+    public ObservableCollection<Vehicle> Vehicles { get; private init; } = new();
+    public ManageVehiclesViewModel(IManageVehiclesService manageVehiclesService)
```
Comment (stewartbebbington) [inline HaulThis/ViewModels/ManageVehiclesViewModel.cs:16]: nit picking but tidy whitespace in this file

File: HaulThis/Views/Misc/AddVechiclePopup.xaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (stewartbebbington) [inline HaulThis/Views/Misc/AddVechiclePopup.xaml:1]: we should probably have a folder for each Employee, Customer and Vehicle popups in this misc folder, will look quite messy when my popups are added as well

**Review by stewartbebbington — CHANGES_REQUESTED** [review]

File: HaulThis/Views/Misc/AddVechiclePopup.xaml (hunk 4 above)
Comment (conor27nash) [reply HaulThis/Views/Misc/AddVechiclePopup.xaml:1]: This might be better to do after all popups are in its own tidy up ticket?

Comment (stewartbebbington) [reply HaulThis/Views/Misc/AddVechiclePopup.xaml:1]: agreed

**Review by stewartbebbington — APPROVED** [review]

**Review by stewartbebbington — APPROVED** [review]

**Review by jamieyoung5 — APPROVED** [review]

**Pull request merged**